<?xml version="1.0"?>
<launch>

<!--  change this depending on which sensor you want to use -->
<!-- 
<arg name = "sensor_type_selected" default = "VLP16" />
<arg name = "sensor_type_selected" default = "HDL32E" />
<arg name = "sensor_type_selected" default = "OUSTEROS0" />
<arg name = "sensor_type_selected" default = "TILT3D" />
<arg name = "sensor_type_selected" default = "TILT4D" /> -->

<!-- try this or the group block -->
<!-- <node pkg= "polar_sensor_models" 	type = "polar_sensor_models_node" name = "polar_sensor_models_node" output="screen" 
if = "$(eval sensor_type_selected == 'VLP16')" /> -->

<!-- group block -->
<!-- <group if= "$(eval arg('sensor_type_selected')==VLP16)">
  <node pkg= "polar_sensor_models" 	type = "polar_sensor_models_node" name = "polar_sensor_models_node" output="screen" >
    <param name="sensorType"  	type="string" value="VLP16"/>
  </node>
</group> -->

<node pkg= "polar_sensor_models" 		type = "polar_sensor_models_node" 	name = "polar_sensor_models_node" 	output="screen" >
		<param name="sensorType"  		type="string" value="VLP16"/>
		<param name="laserDataTopic"  	type="string" value="puck_rear/velodyne_points"/>
		<param name="raysIncl"  		type="int" value="16"/> 				<!-- number of vertical rays  -->
		<param name="inclMin"  			type="double" 	value="-0.26180"/> 		<!-- -15.0° lowest inclination ray (vertical)  -->
		<param name="inclMax"  			type="double" 	value="0.26180"/>		<!-- +15.0° highest inclination ray (vertical) -->
		<param name="inclRes"  			type="double" 	value="0.034907"/>		<!--   2.0° inclination resolution - angle between two vertical rays -->
		<param name="azimMin"  			type="double" 	value="0.0"/>			<!--   0.0° azimuth start angle (horizontal)   -->
		<param name="azimMax"  			type="double" 	value="6.28319"/>		<!-- 360.0° azimuth stop angle (horizontal)    -->
		<param name="azimRes"  			type="double" 	value="0.00349066"/>	<!--   0.2° azimuth resolution - angle between two horizontal rays   -->
<!-- TODO: add firingSeq with YAML file input -->
</node>

</launch>
